bson/json_util.py

```diff
@@ -486,13 +492,13 @@ def _get_dbpointer(doc, dummy):
 
 
 def canonical_object_hook(dct, json_options=CANONICAL_JSON_OPTIONS):
-    keyset = frozenset(dct)
+    keyset = frozenset(key for key in dct if key.startswith('$'))
```

Comment:
The spec says what happens when extra $ keys are present (https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#id43), but what if an extra non-$ keys exist?

Does `'{"$numberInt": "123", "foo": "bar"}'` get parsed into a bson Int32 or a document `{"$numberInt": "123", "foo": "bar"}`?